### DISCUSSION TITLE
Error in post-command-hook

### DISCUSSION THREAD

**Discussion by notuntoward** [joaotavora/eglot#1182] [Q&A]
I've installed `pyright` on Windows 11 from conda-forge, and have installed `eglot` with the following:

```
    (use-package eglot
      :defer t
      :hook (python-mode . eglot-ensure))
```

The same setup works perfectly on MacOS but not on Windows.  On Windows, when I open a python file, I get the error message below:

> Error in post-command-hook (#[0 "\303\304\301\242\305#\210\300\306!\205\0r\211q\210
> ?\205\0\307\310\311 \")\207" [#<buffer sutil.py> (#0) eglot--managed-mode remove-hook post-command-hook nil buffer-live-p apply eglot--connect eglot--guess-contact] 4]): (void-function project-name)

No `eglot` commands work at this point.  Any ideas?

**Comment by joaotavora** [comment]
>  Any ideas?

Read https://joaotavora.github.io/eglot/#Starting-Eglot about the use of `eglot-ensure`. Don't use it unless you know M-x eglot works.

Then read the troubleshooting section of the same manual. 

It would seem you are missing an Eglot dependency.

**Comment by wildsebastian** [comment]
I have the same issue. You can fix it by copying [this](https://github.com/emacs-mirror/emacs/blame/403bad2f604ab096fbf04dad39725198e8c59e73/lisp/progmodes/project.el#L279) function into your emacs config.
I don't have this yet in project.el in my emacs 28.2 version. This was introduced into eglot with [this](https://github.com/joaotavora/eglot/commit/fb8111d8222b09641a6aaab02d846ceac3ae97ed) commit.

**Reply by joaotavora** [reply]
This is a bad idea. 

The better idea is to discover why your Eglot installation is corrupt and didn't pull in the correct version of the `project.el` package.  It should, as long as you install Eglot using `M-x package-install`

**Reply by wildsebastian** [reply]
Good point! I noticed it and was in need of a quick fix.
Thank you for the hint. I am not using `M-x package-install`, but nix to install my emacs packages. Based on that everything should be correct. But as soon as I can I will dig deeper on that.
